libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/AudioFocusDelegate.kt

```diff
@@ -1,8 +1,18 @@
 package com.mapbox.navigation.ui.voice.api
 
-internal interface AudioFocusDelegate {
+/**
+ * An Api that allows you to interact with the audio focus
+ */
+interface AudioFocusDelegate {
```

Comment:
Could you clarify why this need to be public? Are we planning to support `requestFocus` / `abandonFocus` APIs to be called from the client? Is that going to impact the internal calls somehow? 🤔

Reply:
Yes, we need to make our implementation for AudioFocusDelegate and pass it to `MapboxVoiceInstructionsPlayer` as default

Reply:
```kotlin
voiceInstructionsPlayer = MapboxVoiceInstructionsPlayer(
            this,
            getMapboxAccessTokenFromResources(),
            Locale.US.language,
            audioFocusDelegate = object : AudioFocusDelegate {
                override fun requestFocus(): Boolean {
                    TODO("Not yet implemented")
                }

                override fun abandonFocus() {
                    TODO("Not yet implemented")
                }

            }
        )
```